packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ClientProvider.cs

```diff
@@ -160,19 +161,25 @@ protected override FieldProvider[] BuildFields()
                 }
             }
 
-            // Add optional client parameters as fields
-            foreach (var p in _inputClient.Parameters)
+            // Add optional client parameters and onClient: true parameter from operations as fields
+            var allParameters = _inputClient.Parameters.Concat(_inputClient.Operations.SelectMany(op => op.Parameters).Where(p => p.Kind == InputOperationParameterKind.Client)).DistinctBy(p => p.Name);
```

Comment:
we have twice places getting this `allParameters`.
Could we make this a field `_allClientParameters` in this `ClientProvider` class?

Reply:
Yes. it is used in two places. and it can be use an field to store this information. but I was concerned that it will occupied the memory all the time during the whole generation. If we use the local variable which may be released when it is not used.